deepomatic/dmake/deepobuild.py

```
@@ -923,10 +923,12 @@ def _get_service_(self, service):
         raise DMakeException("Could not find service '%s'" % service)
 
     def _get_link_opts_(self, commands, service):
-        if common.options.dependencies and service.tests.has_value():
-            docker_links_names = service.tests.docker_links_names
-            if len(docker_links_names) > 0:
-                append_command(commands, 'read_sh', var = 'DOCKER_LINK_OPTS', shell = 'dmake_return_docker_links %s %s' % (self.app_name, ' '.join(docker_links_names)), fail_if_empty = True)
+        if common.options.dependencies:
+            needed_links = service.needed_links
+            if service.tests.has_value(): # deprecated
+                needed_links += service.tests.docker_links_names
+            if len(needed_links) > 0:
```

maybe raise an error if both new and deprecated fields are used in parallel?